Added development section to the readme (#44)
OSIPI/osipi: README.md

Reviewer: I can see you've added instructions on setting up the development environment. Perhaps there is good reason for this, but shouldn't these instructions be merged into the developer guide?

Author: I think keeping the development environment set in the README for now is important, as it informs anyone coming to the repo on how to start contributing. Once we have a published version, we can update the README accordingly and remove the development setup part. What do you think, @ltorres6 ?

- Push your changes to your fork.
- Submit a pull request to the main OSIPI repository.

## Development Setup

There are two ways to set up your development environment for this project. You can choose the one that suits you best.

### Option 1: Using venv and requirements.txt

1. Clone the repository to your local machine.
2. Navigate to the project directory.
3. Create a virtual environment using `venv` by running the following command:

    ```bash
    python -m venv venv
    ```

4. Activate the virtual environment:

    - On Windows, run:

        ```bash
        .\venv\Scripts\activate
        ```

    - On Unix or MacOS, run:

        ```bash
        source venv/bin/activate
        ```

5. Install the required packages from `requirements.txt`:

    ```bash
    pip install -r requirements.txt
    ```

> ### Option 2: Using Poetry

Reviewer 2: We want to make poetry a requirement. It's small overhead and it ensures packages won't be missing/incompatible. Remove this and the associated requirements.txt please.

1. If you don't have Poetry installed, you can install it by following the instructions on the [official Poetry website](https://python-poetry.org/docs/#installation).
2. Clone the repository to your local machine.
3. Navigate to the project directory.
4. Install the project dependencies with Poetry:

    ```bash
    poetry install
    ```

5. Activate the Poetry environment:

    ```bash
    poetry shell
    ```

## Setting up pre-commit

`pre-commit` is a tool that we use to maintain high-quality code in this project. It runs checks (hooks) on your code each time you commit changes. Here's how to set it up:

1. In the project directory, run the following command to install the `pre-commit` hooks:

    ```bash
    pre-commit install
    ```

2. You can run all pre-commit hooks on all files with:

    ```bash
    pre-commit run --all-files
    ```

After this, `pre-commit` will automatically run the hooks each time you commit changes to the project. If any of the hooks fail,
you'll need to fix the issues and then try committing your changes again.

> Also some of hooks are auto fixed if you see file was modified after running `pre-commit run --all-files` then you can add the changes and commit again.

Reviewer 2: ```suggestion
NOTE: Some of hooks automatically fix/format your files to specific standards. If you see file was modified after running `pre-commit run --all-files` then you'll need to add the changes (git add) and commit again.
```

## Development Setup

> We used poetry to manage the dependencies for this project.

Reviewer 2: ```suggestion
We use poetry to manage the dependencies for this project.
```

    ```bash
    pre-commit run --all-files
    ```
**NOTE:**

> - Next times you commit changes, `pre-commit` will run the hooks automatically.

Reviewer 2: ```suggestion
- Next time you commit changes, `pre-commit` will run the hooks automatically.
```

> - Some of hooks automatically fix/format your files to specific standards. If you see that some of your files have been changed after a commit, don't worry! It's just `pre-commit` doing its job.

Reviewer 2: ```suggestion
- Some hooks automatically fix/format your files to specific standards. If you see that some of your files have been changed after a commit, don't worry! It's just `pre-commit` doing its job.
```

Reviewer 2: Nice.
